### DISCUSSION TITLE
Running multiple versions of the same app?

### DISCUSSION THREAD

**Discussion by Relaxe111** [static-web-server/static-web-server#214] [Q&A] [answered]
Hi, I would like to ask if it is possible to make it running multiple versions of the same app simultaneously?
for example i have v1 and v2, located dist/v1 dist/v2 and when i visit my.site.com/v1/ shall point to v1 but my.site.com/v2/ to v2
Is it possible?

**Comment by joseluisq** [comment]
I did not get the question at all. 
But If you have two web apps `dist/v1`, `dist/v2` and you want to put them under the same `my.site.com` domain. Then just serve your content from the parent directory, like so:

```sh
static-web-server --port 80 --root dist/
```

Now if you try to reach the websites then you should be able to navigate to `my.site.com/v1` or `my.site.com/v2`.

**Comment by Relaxe111** [comment]
Hi thank you for your answer. Sorry for my bad english :(.

I think you answered partly to my question. 

I want to run multiple versions of the same app simultaneously from the same domain, And i am researching a way how  can I do so? 

I figured out how to build static assets and put them under same dist/* folder, incrementally.

Now I am looking for possibility to access those different versions simultaneously. 

I think on few options:

option 1 - through the path like my.site.com/v1 or /v2.
option 2 - through headers like x-myapp-version: v2.


 
If i understood correctly with SWS i could do only option 1? 

Or is possible to do option 2 as well?

can i set a default route if one tries to access / path?
for example: if i try to access my.site.com i would like to be redirected to a default path, where i can specify somewhere in SWS configuration, like /v2, and to be served my.site.com/v2

**Reply by joseluisq** [reply]
> option 2 - through headers like x-myapp-version: v2.
> If i understood correctly with SWS i could do only option 1?
> Or is possible to do option 2 as well?

I don't get the second option. But SWS can do the first one.

> can i set a default route if one tries to access / path?
> for example: if i try to access my.site.com i would like to be redirected to a default path, where i can specify somewhere in SWS configuration, like /v2, and to be served my.site.com/v2

You could redirect from `my.site.com/` to `my.site.com/v2` via the config file like this:

```toml
# ...
[advanced]

### URL Redirects
[[advanced.redirects]]
source = "/"
destination = "/v2"
kind = 302
```

See [URL redirects](https://static-web-server.net/features/url-redirects/) for more details.

**Comment by Relaxe111** [comment]
Related to option 2: We are running couple of micro front ends which are combined in browser when user accesses our site. We have a master app  (which basically is a router) which mount or unmount child apps.  So When i call from master app  child apps i can pass headers to  http call, like: `get: https://my-child.site.com  "header x-myapp-version: v2"`, and if SWS could somehow read the header and serve `dist/v2` which matches x-myapp-version `v2`. Is that possible to achieve with SWS?

**Reply by joseluisq** [reply]
Basically, you want to control SWS root serving directory based on headers. This is not possible.
Because SWS is a static web server which means you start the server by providing a root directory and that root directory can not be changed at runtime. What can change only is the content (directories or files) inside the root.
If you want to change the root directory then you have to restart the server.

**Comment by Relaxe111** [comment, marked as answer]
I've managed to make it work with help of paths.
now my folder structure looks like:
```
public:
    mf-chat/
        0.0.1/
        0.0.3/
        etc...
    mf-calendar/
        0.0.2/
        0.0.3/
        0.3.2/
        etc...
```
and i can call these like `https://cdn.mysite.com/mf-chat/0.0.1/`

**Reply by joseluisq** [reply]
Great, as an addition you can even customize your routes using [URL Rewrites](https://static-web-server.net/features/url-rewrites/).

**Comment by Relaxe111** [comment]
closing
